### ISSUE TITLE
Uncaught TypeError: Cannot read properties of undefined (reading 'content')

### ISSUE THREAD

**Issue by nik32** [fkhadra/react-toastify#858]
**Do you want to request a *feature* or report a *bug*?**
bug report

**What is the current behavior?**
So is complicated to explain as this might be specific to my project, But I will try!! In our product - we are having a modal on top of modal. Both of these modals are having their own ToastContainer [with multiContainer enabled and containerId provided to both - so that toast produced in one modal is not shown in the other one]. **Now If after closing a toast - I 'very quickly' close the modal, I am getting the below error -** 
<img width="1418" alt="image" src="https://user-images.githubusercontent.com/51018572/196342010-43ca0ae3-9bd4-4bca-8a24-a76b1b5e3061.png">

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React. Paste the link to your CodeSandbox (https://codesandbox.io/s/new) example below:**
Steps to reproduce are given above. It will be time consuming to give the demo at the moment. But let me know if you are not able to reproduce the error with the above steps or if you really want the demo - I will try and provide a CodeSandbox.

**What is the expected behavior?**
The error should not occur coz it then stoped any other toasts to be displayed on the page till we refresh the page. NOTE - the first time I tested, other toasts were not being shown after the error. But in sub-sequest testing I have found that other toasts are not being stopped from displaying.

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**
React 18 [with TS and NextJS]. I have tested on 2 browser - a. Microsoft edge on MacOS and b. safari. On both I have been able to to reproduce it with the steps told above. As edge is chromium based - i think this should also happen with chrome, don't know about firefox. On safari the following was the error [this might give you more context about this bug] - 
<img width="833" alt="image" src="https://user-images.githubusercontent.com/51018572/196344547-69946200-3640-408a-a4cd-8585cc21e4ae.png">

**Comment by erikbg7** [comment]
Looks like a bug caused by unmounting the ToastContainer with the Toast before removing it correctly.

The main issue is that we expect the toast to exist but it does not:
https://github.com/fkhadra/react-toastify/blob/b53938a61756c7fc1d3de47eaba858fc2691e520/src/hooks/useToastContainer.ts#L191

So it breaks when we try to access it:
https://github.com/fkhadra/react-toastify/blob/b53938a61756c7fc1d3de47eaba858fc2691e520/src/utils/mapper.ts#L5

Could be easily solved if it is considered a bug, but you could probably fix it just by moving the ToastContainers out of the Modals.

**Comment by nik32** [comment]
Hey @erikbg7, thanks for taking time to look into this issue!!

> Could be easily solved if it is considered a bug, but you could probably fix it just by moving the ToastContainers out of the Modals.

That is actually a bit problematic in my case - as I have my own custom modal. If I move the ToastContainers out of my modals - the toast comes behind the backdrop of modal. Thus not allowing the user to close the toast or see the toast properly.

**Comment by nettum** [comment]
I can also confirm that I run into this, probably when using `React.useRef()` and `toast.dismiss`. 
I only have one `ToastContainer` in a layout wrapper component and the toast itself is fired from inside a `<Portal>`-component from `react-portal`. Have not done enough testing yet to figure out the root cause and get this error consistent.

Example:
```
const toastId = React.useRef<Id | null>(null);

toastId.current = toast("some text" { isLoading: true });

if (someCondition) {
  toast.update(toastId.current, { ... });
} else {
  toast.dismiss(toastId.current);
}
```


```
Ubuntu 22.04.1 LTS
Google Chrome 106.0.5249.103
react-toastify 9.0.8
react 17.0.2
```

**Comment by nik32** [comment]
For me, in the modal situation, the root cause most likely is to be "closing the modal too quickly", after dismissing the toast [or in simpler terms - when the time b/w closing the modal and toast is very less']. I was able to produce this error every time doing so.

**Comment by nettum** [comment]
Thanks for the info @nik32! For the record, I was now also able to get the same error somewhere else in the app without any modal context, so there might be something else going on with mouting / unmounting and "quick navigation".

**Comment by nettum** [comment]
I had the ToastContainer in a layout component that sometimes had to do re-renders. Will move it further out to see if the problem disappears.

**Comment by nik32** [comment]
@erikbg7 any update on this?

**Comment by baxgas** [comment]
Same problem here, using the ToastContainer in a refine + next.js app, have to put it in a layout, so it can handle globally the toast messages.

**Comment by nettum** [comment]
FWIW this was solved for me after putting the ToastContainer inside the `_app.tsx` instead of our own layout file. Not gotten any errors after that change.

**Comment by baxgas** [comment]
Thanks, @nettum I tried it, put it outside the `<Refine>` wrapper but still have the same issue :-( 
The strange thing is that it works fine with some pages, eg. `create` resource works, but `edit` resource gives this Unvaught TypeError, albeit they use the same React component, with this `onSubmit` trigger
```
    const onSubmit = async (formData: any) => {
        const data = onFinish(formData)
        data.then((res) => {
            if (res) {
                toast(`${res?.data?.name} succesfully ${isEdit ? 'edited' : 'created'}`, {
                    type: 'success',
                })
            } else {
                toast('Something went wrong', {
                    type: 'error',
                })
            }
        })
    }
```

**Comment by baxgas** [comment]
I found a workaround for it, to wrap it in setTimeout, with some minimal (eg. 100) value

**Comment by Envoy-VC** [comment]
I was having the same error 
`TypeError: Cannot read properties of undefined (reading 'content')`

I resolved it by adding a small delay in the notification appearance

eg-

```js
const sendTransaction = () => {
    toast.promise(
      writeAsync,
      {
        pending: "Promise is pending",
        success: "Promise resolved 👌",
        error: "Promise rejected 🤯",
      },
      {
        position: toast.POSITION.BOTTOM_LEFT,
        delay: 100,
      }
    );
  };
```

**Comment by igor17400** [comment]
Any solution on this?
I'm facing the same problem

**Comment by Envoy-VC** [comment]
> I was having the same error `TypeError: Cannot read properties of undefined (reading 'content')`
> 
> I resolved it by adding a small delay in the notification appearance
> 
> eg-
> 
> ```js
> const sendTransaction = () => {
>     toast.promise(
>       writeAsync,
>       {
>         pending: "Promise is pending",
>         success: "Promise resolved 👌",
>         error: "Promise rejected 🤯",
>       },
>       {
>         position: toast.POSITION.BOTTOM_LEFT,
>         delay: 100,
>       }
>     );
>   };
> ```

@igor17400 did you try this?

**Comment by berci-i** [comment]
I also have this problem, and the main issue is that it doesn't give me a traceback (so I am not sure where in the code this occurred ). 

I think at the very least rect-toastify should provide the traceback to let you know what caused the issue.

Thank you!

**Comment by nik32** [comment]
@erikbg7 @fkhadra since so many people are facing this same issue... can't this now be considered a bug and be solved (especially when you said that this is easy to solve also)??

**Comment by fkhadra** [comment]
Would it be possible to share a codesanbox with the issue? This would speedup the investigation

**Comment by kiyan-alav** [comment]
Hi I have the same error :/
{
Uncaught TypeError: Cannot read properties of undefined (reading 'content')
    at y (mapper.ts:5:1)
    at deleteToast (useToastContainer.ts:160:1)
}
I put my toast in 2 route in my home route and products route. when I switch route to route that calculate the sum of products, my app crash and face this error.
any solution on this?

**Comment by nik32** [comment]
@fkhadra is https://github.com/fkhadra/react-toastify/issues/858#issuecomment-1284292042 enough for you to understand the issue or you would need the reproduction?? In my case it would be rather time taking to reproduce it in codesandbox. @nettum @baxgas @Envoy-VC @berci-i @igor17400 @kiyan-alav is your case easy to reproduce on codesandbox??

**Comment by fkhadra** [comment]
At least a repository would be super helpful

**Comment by GVVC** [comment]
> I was having the same error `TypeError: Cannot read properties of undefined (reading 'content')`
> 
> I resolved it by adding a small delay in the notification appearance
> 
> eg-
> 
> ```js
> const sendTransaction = () => {
>     toast.promise(
>       writeAsync,
>       {
>         pending: "Promise is pending",
>         success: "Promise resolved 👌",
>         error: "Promise rejected 🤯",
>       },
>       {
>         position: toast.POSITION.BOTTOM_LEFT,
>         delay: 100,
>       }
>     );
>   };
> ```

**Comment by GVVC** [comment]
Hi, tried this but facing same error,

**Comment by karvas** [comment]
> I found a workaround for it, to wrap it in setTimeout, with some minimal (eg. 100) value

This is working for me too! The error has stopped after I added a delay like this:

`toast.success('Successful sign up.', {delay: 100});`

**Comment by nik32** [comment]
@fkhadra sorry for the late reply. I tested and I was not able to reproduce it in my case. If you like you can ask other also who were facing this issue or else you can close this

**Comment by kevin-dev71** [comment]
i upgraded to the version 9.1.3 and the problem its solved
